src/assets/ts/definitions/basics.ts

```diff
@@ -255,6 +255,38 @@ keyDefinitions.registerAction(new Action(
   },
 ));
 
+keyDefinitions.registerAction(new Action(
+  'visual-line-join',
+  'Join all selected rows',
+  async function({ session, visual_line }) {
+    if (visual_line == null) {
+      throw new Error('Visual_line mode arguments missing');
+    }
+
+    if (visual_line.num_rows >= 1) {
+      try {
+        const resultArr = await Promise.all(
+          visual_line.selected.map(async (path) => {
+            return await session.getTextRecusive(path);
+          })
+        );
+
+        const resultArrClear = resultArr.map(function(x) { return x.replace(/(?:\r)/g, ''); });
```

Comment:
what does this regex do?  could you leave a comment as well?

Reply:
This regex removes \r (Carriage Return) from each line.

Reply:
cool, could be good to do in visual-line-join instead as well, what do you think?

Reply:
> cool, could be good to do in visual-line-join instead as well, what do you think?

I did not understand, could you explain?

Reply:
nevermind, seems fine, just leave a comment!  when does the carriage return exist in the text?  (I'm used to unix-like OS so don't know much about it)

Reply:
If I copy text and then paste text from clipboard on Windows machine, in each line there are \r\n. And it looks like an additional char at the end.
![image](https://user-images.githubusercontent.com/3366415/80285762-d41b6800-872f-11ea-8b35-711900aeb07e.png)

Reply:
cool - could be good to fix that in the pasting function too!  i don't mind leaving it here for now also though

Reply:
I have not reached the paste function yet :)
There is also an unpleasant bug. When the search mode is open, it is impossible to insert text into the search bar; instead, it is inserted at the cursor position on the previous screen.

Reply:
ah you mean pasting into the search bar doesn't work?  good catch

Reply:
Yes, I tried to paste text into the search bar and nothing worked. After I closed the search mode, all that I pasted turned out to be in the line where the cursor was.

Reply:
ok opened this issue https://github.com/WuTheFWasThat/vimflowy/issues/338